### PR TITLE
fix: remove gap cursor

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

A lot of people don't understand what this is. Chatted about in https://posthog.slack.com/archives/C05N9R3HT7V/p1696434085225909

## Changes

- Remove it entirely

## How did you test this code?

https://github.com/PostHog/posthog/assets/6685876/da3669b9-9782-4a88-b9b8-20f5e707fcfe

### CODE REVIEW

File: frontend/src/scenes/notebooks/Notebook/Editor.tsx [hunk 1]
```diff
@@ -65,6 +65,7 @@ export function Editor({
             CustomDocument,
             StarterKit.configure({
                 document: false,
+                gapcursor: false,
```
Comment (benjackwhite) [inline frontend/src/scenes/notebooks/Notebook/Editor.tsx:68]: I don't think this is mergeable. I find the weird gap cursor preferable to not being able to place a cursor between two nodes...

Not against this change, as long as we do the extra work to make sure you can add a line between nodes when we click there. Might be best done by detecting the selection and then inserting a line if there isn't one (not sure how hacky that would be)

**Review by benjackwhite — CHANGES_REQUESTED** [review]

**Comment by daibhin** [comment]
@benjackwhite being honest I actually like gap cursors, but at least three people were confused by them. I used to reference Linear as an example of a tool that had a similar behaviour but when I went to reply in Slack it looks like they have removed theirs now too:

https://github.com/PostHog/posthog/assets/6685876/76e2b25b-28ca-4ad9-bf94-9e3a04f09f20
(No indication that the selection is between the two images after pressing the right arrow key or clicking between them until you type or press enter)

Maybe the industry is moving away from them as a concept 🤷 My personal vote is to leave the gap cursors alone

**Comment by benjackwhite** [comment]
> @benjackwhite being honest I actually like gap cursors, but at least three people were confused by them. I used to reference Linear as an example of a tool that had a similar behaviour but when I went to reply in Slack it looks like they have removed theirs now too:
> 
> https://github.com/PostHog/posthog/assets/6685876/76e2b25b-28ca-4ad9-bf94-9e3a04f09f20 (No indication that the selection is between the two images after pressing the right arrow key or clicking between them until you type or press enter)
> 
> Maybe the industry is moving away from them as a concept 🤷 My personal vote is to leave the gap cursors alone

@daibhin - my problem isn't with the cursor itself, but rather with the fact that this change actually removes the ability to place your cursor between two nodes 😅 Which is the bit I don't agree with. We should either fix that, or simply change this to hide the cursor when it is in a gap.

**Review by benjackwhite — APPROVED** [review]
Now this is slick 👍

**Pull request merged**